packages/jobs/lib/crons/reconcileTemporalSchedules.ts

```diff
@@ -0,0 +1,103 @@
+import * as cron from 'node-cron';
+import { telemetry, LogTypes, LogActionEnum, errorManager, ErrorSourceEnum, SyncClient, db, getRunningSchedules } from '@nangohq/shared';
+import { getLogger, metrics } from '@nangohq/utils';
+import tracer from 'dd-trace';
+
+const logger = getLogger('Jobs.TemporalSchedules');
+
+export function reconcileTemporalSchedules(): void {
+    cron.schedule('*/30 * * * *', async () => {
+        const start = Date.now();
+        try {
+            await exec();
+        } catch (err: unknown) {
+            const e = new Error('failed to reconcile temporal schedules');
+            e.cause = err instanceof Error ? err.message : err;
+            errorManager.report(e, { source: ErrorSourceEnum.PLATFORM }, tracer);
+        }
+        metrics.duration(metrics.Types.RENCONCILE_TEMPORAL_SCHEDULES, Date.now() - start);
+    });
+}
+
+export async function exec(): Promise<void> {
+    logger.info('[reconcileTemporalSchedules] starting');
+
+    const today = new Date();
+    const lockKey = parseInt(`2${today.getFullYear()}${today.getMonth() + 1}${today.getDate()}`);
```

Comment:
is the lock to prevent multiple instance of the cron to execute at the same time? What if the last cron of a day takes a long time or never finishes?

Reply:
I added this on my cron to avoid multiple instances conflict yes. Not sure I get the last part of the message

Reply:
the key is the date right? so 2 cron instances started on different day could run at the same time, no?

Reply:
good point indeed, on my side I used a constant number. Ideally here we should crc32 a constant string

Reply:
Updated with https://github.com/NangoHQ/nango/pull/2149/commits/675bc605697e650fa71a99b63b439051408a2171